charts/argo-rollouts/templates/controller/deployment.yaml

```
@@ -78,6 +78,9 @@ spec:
           {{- toYaml .Values.controller.readinessProbe | nindent 10 }}
         securityContext:
           {{- toYaml .Values.containerSecurityContext | nindent 10 }}
+        {{- if .Values.controller.lifecycle }}
+        lifecycle: {{ toYaml .Values.controller.lifecycle | nindent 10 }}
+        {{- end }}
```

nits; `with` is kinda DRY ?
```suggestion
        {{- with .Values.controller.lifecycle }}
        lifecycle: {{ toYaml . | nindent 10 }}
        {{- end }}
```